mp2-common/src/digest.rs

```diff
@@ -138,14 +139,16 @@ impl SplitDigestTarget {
             multiplier: digest_mul,
         }
     }
-    /// Recombine the split and individual targets into a single one. It hashes the individual
-    /// digest first as to look as a single table.
+    /// First compute the individual row digest of each component (i.e. digesting again to make a
+    /// digest of a row). Then recombine the split and individual targets into a single one. It
+    /// hashes the individual digest first as to look as a single table.
     /// NOTE: it takes care of looking if the multiplier is NEUTRAL. In this case, it simply
     /// returns the individual one. This is to accomodate for single table digest or "merged" table
     /// digest.
     pub fn cond_combine_to_row_digest(&self, b: &mut CBuilder) -> DigestTarget {
-        let digest_ind = b.map_to_curve_point(&self.individual.to_targets());
-        cond_circuit_hashed_scalar_mul(b, self.multiplier, digest_ind)
+        let row_digest_ind = b.map_to_curve_point(&self.individual.to_targets());
+        let row_digest_mul = b.map_to_curve_point(&self.multiplier.to_targets());
```

Comment:
How can `cond_circuit_hashed_scalar_mul` works if we wrap `self.multiplier` with another digest? `cond_circuit_hashed_scalar_mul` is checking whether `multiplier == NEUTRAL`, but now since we are providing `D(self.multiplier)` to this method, then `multiplier == NEUTRAL` inside `cond_circuit_hashed_scalar_mul` will never be true. What am I missing?